### PR TITLE
Created ask-sdk-model-runtime package

### PR DESCRIPTION
**Pull request by ShenChen93**
This package contains the defintions of LWA servies, implementations of ApiClient and its componenets. This package can be reused across multiple models released by ask sdk team in future, also the current ask-sdk-models (Node.js) can add this as dependency.

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: ask-sdk-model-runtime/README.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+ask-sdk-model-runtime is the common models package by
+the Software Development Kit (SDK) team for Node.js that
+provides common implementations like ApiClient
+BaseServiceClient, Authentication Configuration etc.
+that is used by the SDK models.
+
+## Installing
+
+```
+npm install --save smapi-sdk-model-runtime
```
Comment (nikhilym) [inline ask-sdk-model-runtime/README.md]: Do you mean `ask-sdk-model-runtime`?

File: ask-sdk-model-runtime/package.json [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    },
+    "author": "Amazon.com",
+    "contributors": [
+      {
+        "name": "ASK SDK",
+        "email": "ask-sdk@amazon.com"
+      }
+    ],
+    "license": "Apache-2.0",
+    "keywords": [
```
Comment (nikhilym) [inline ask-sdk-model-runtime/package.json:18]: You should probably include models and model runtime in this list

File: ask-sdk-model-runtime/index.ts [hunk 3]
```diff
@@ -0,0 +1,480 @@
...
+        if (isQueryStart) {
+            sb.push('&');
+        } else {
+            sb.push('?');
+        }
+
+        params.forEach((obj) => {
+            sb.push(encodeURIComponent(obj.key));
+            sb.push('=');
+            sb.push(encodeURIComponent(obj.value));
```
Comment (nikhilym) [inline ask-sdk-model-runtime/index.ts:241]: Will this work with multiple query values for the same query parameter?

File: ask-sdk-model-runtime/README.md (hunk 1 above)
Comment (ShenChen93) [reply ask-sdk-model-runtime/README.md]: Good catch!

File: ask-sdk-model-runtime/package.json (hunk 2 above)
Comment (ShenChen93) [reply ask-sdk-model-runtime/package.json:18]: Got it, will add

File: ask-sdk-model-runtime/index.ts (hunk 3 above)
Comment (ShenChen93) [reply ask-sdk-model-runtime/index.ts:241]: Yes!

**Review by nikhilym — APPROVED** [review]
LGTM. 🚢

**Pull request merged**